block/src/Pos/Block/Network/Retrieval.hs

```diff
@@ -79,10 +79,9 @@ retrievalWorkerImpl
     :: forall ctx m.
        (BlockWorkMode ctx m)
     => Timer -> SendActions m -> m ()
-retrievalWorkerImpl keepAliveTimer SendActions {..} =
-    handleAny mainLoopE $ do
-        logInfo "Starting retrievalWorker loop"
-        mainLoop
```

Comment:
This was fixed in the earlier (unmerged) PR: https://github.com/input-output-hk/cardano-sl/pull/2478
The exception handler only went on the first iteration.